packages/dd-trace/src/esm-hook.js

```diff
@@ -0,0 +1,43 @@
+'use strict'
+
+const parse = require('module-details-from-path')
+
+let iitm
+
+function esmHook (instrumentedModules, hookFn) {
+  if (!iitm) {
+    // TODO(bengl) why does this need to be loaded asynchronously???
+    Promise.resolve().then(() => {
+      iitm = require('import-in-the-middle')
```

Comment:
Why lazy-load this? It doesn't have any side-effects, as far as I can tell. 🤔

Reply:
As mentioned in the TODO, I'm not really sure!

If I simply require it like any other module, the integration tests I added for ESM will fail (default exports are broken). I'm a bit perplexed by this to be honest. I'll look into it further when I have a chance.

This async load actually prevents `await import()` from being instrumented properly in CJS. I don't know if this tradeoff is _necessary_, but thought it might be reasonable for a first pass at this.

Reply:
Figured it out. I was dealing with default exports incorrectly. Upcoming push will fix it (and remove the tradeoff).